doc/specs/stdlib_math.md

```diff
@@ -649,3 +649,84 @@ program demo_math_all_close
     
 end program demo_math_all_close
 ```
+
+### `diff`
+
+#### Description
+
+Computes differences between adjacent elements of an array.
+
+#### Syntax
+
+For rank-1 array
+```fortran
+y = [[stdlib_math(module):diff(interface)]](x [, n, prepend, append])
+```
+and for rank-2 array
+```fortran
+y = [[stdlib_math(module):diff(interface)]](x [, n, dim, prepend, append])
+```
+
+#### Status
+
+Experimental.
+
+#### Class
+
+Pure function.
+
+#### Arguments
+
+Note: The `x`, `prepend` and `append` arguments must have same `type`, `kind` and `rank`.
+
+`x`: Shall be a `real/integer` and `rank-1/rank-2` array.
+This argument is `intent(in)`.
+
+`n`: Shall be an `integer` scalar.
+This argument is `intent(in)` and `optional`, which is `1` by default.
+It represents to calculate the n-th order difference.
+
+`dim`: Shall be an `integer` scalar.
+This argument is `intent(in)` and `optional`, which is `1` by default.
+It represents to calculate the difference along which dimension.
```

Comment:
This sentence sounds strange to me. What do you mean?

Reply:
[numpy.diff](https://numpy.org/doc/stable/reference/generated/numpy.diff.html):
> `axis`: `int`, `optional`
The axis along which the difference is taken, default is the last axis.

[matlab.diff](https://ww2.mathworks.cn/help/matlab/ref/diff.html?lang=en):
> `Y = diff(X,n,dim)` is the nth difference calculated along the dimension specified by `dim`. The dim input is a positive integer scalar.

Reply:
Perhaps change the last sentence to: `It represents the dimension along which the difference is calculated.`